src/Vault.php

```diff
@@ -44,12 +53,37 @@ public function unseal() {
     }
 
     public function get($key) {
-      $this->unseal();
-      $response = $this->data()->get($key);
-      return json_decode($response->getBody())->data->value;
+        $response = $this->getAuthClient()->get($key);
+        return $this->extractSecrets($response);
+    }
+
+    private function extractSecrets($response)
+    {
+        $data = json_decode($response->getBody())->data;
+
+        if (property_exists($data, 'secrets')) {
+            return $data->secrets;
+        }
+
+        return $data->value;
+    }
+
+    public function getAuthClient()
+    {
+        if ($this->method != null) {
+            return $this->factory()->get($this->method);
+        }
+
+        return $this->data();
     }
 
     public function write($key, $value) {
-      $this->data()->write($key, ['value' => $value]);
+        $arrayKey = 'value';
+
+        if ($this->method == 'github') {
+            $arrayKey = 'secret';
+        }
```

Comment:
This is for not breaking the current projects that use it, right?

Reply:
Yes :D